ast/src/analyzed/mod.rs

```diff
@@ -1940,11 +1938,7 @@ mod tests {
         let mut pil_result = Analyzed::default();
         pil_result.append_polynomial_identity(AlgebraicExpression::Number(0), SourceRef::unknown());
         pil_result.append_polynomial_identity(AlgebraicExpression::Number(5), SourceRef::unknown());
-        if let Identity::Polynomial(PolynomialIdentity { id, .. }) = &mut pil_result.identities[1] {
-            *id = 6;
-        } else {
-            panic!();
-        }
```

Comment:
why?

Reply:
This code was there to match the IDs, but now IDs are always dense (because of [this call](https://github.com/powdr-labs/powdr/blob/9eac36701e45f368094acb667d2c538017701231/ast/src/analyzed/mod.rs#L334) to `reallocate_identity_ids`). This caused a test failure.

The change above is also needed to fix this test, otherwise we'd have to manually make the IDs of `pil_result` start from 0 here.